yarpl/include/yarpl/flowable/EmitterFlowable.h

```diff
@@ -164,8 +161,12 @@ class EmiterSubscription : private Subscription, private Subscriber<T> {
       int64_t emitted;
       bool done;
 
-      std::tie(emitted, done) =
-          emiter_->emit(get_ref<Subscriber<T>>(this), current);
+      std::tie(emitted, done) = emiter_->emit(this_subscriber, current);
+
+      if(this->count() == 1) {
```

Comment:
We have one more while/true loop below. We might put this check to there? Are we sure that we will not stuck in that loop?